### DISCUSSION TITLE
overview/hello-world

### DISCUSSION THREAD

**Discussion by giscus** [itwanger/tobebetterjavaer-giscus#191] [Announcements]
# overview/hello-world

Java程序员进阶之路，小白的零基础Java教程，编写第一个Java程序：Hello World

https://tobebetterjavaer.com/overview/hello-world.html

**Comment by Khaoden** [comment]
二哥，这里我有一个疑问，上面说java具有解释和编译，但因为JIT是后来引入的，所以大家称java编译性语言（先入为朱），可是JIT是编译好再运行，它是后来的，先入的是不是解释嘛

**Reply by guojianjun-ts** [reply]
我的理解是，在JIT没出现之前，java语言只是按部就班的 先用javac编译为字节码，再靠JVM对字节码进行逐条解释。这是线性的
但是JIT出现后，改变了这个效率低下的情况，因为在执行过程中使用了逐条解释执行和JIT即时编译两种方式共存的混合方式。这是混合的。
可能把二哥说的话变成:Java是编译与解释并存的语言，你就好理解了
